Assets/Scripts/UI/PickupPup/DogProfile/DogShelterProfile.cs

```diff
@@ -24,6 +25,7 @@ protected override void setReferences()
     {
         base.setReferences();
         iconsObject.SetActive(false);
+        adoptUIButton = adoptButton.GetComponent<UIButton>();
```

Comment:
This seems redundant. There should just be a single ref `UIButton adoptButton` and you should route any calls you need to the `Button` class through our custom `UIButton` class